### PR TITLE
add CedarMetadataRecord.clean

### PR DESCRIPTION
**Pull request by aaxelb**
## Purpose
avoid headaches trying to make sense of silent errors after they've happened

## Changes
add `CedarMetadataRecord.clean` to validate the record json using the template's jsonschema

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

### CODE REVIEW

**Comment by cslzchen** [comment]
Just kicked off a re-run of `api3_and_osf`, the one test failed randomly on one of my PR as well.

File: osf/models/cedar_metadata.py [hunk 1]
```diff
@@ -43,3 +44,10 @@ def get_template_name(self):
     def save(self, *args, **kwargs):
         self.guid.referent.update_search()
         return super().save(*args, **kwargs)
+
+    def clean(self):
+        super().clean()
+        _cedar_template_jsonschema = self.template.template
+        _cedar_record_json = self.metadata
+        # raises django.core.exceptions.ValidationError when invalid
+        JsonschemaValidator(_cedar_template_jsonschema)(_cedar_record_json)
```
[the remark this answers is not in the document]
Comment (adlius) [reply osf/models/cedar_metadata.py:53]: I don't think you can verify this on the mirage server.

[the remark this answers is not in the document]
Comment (cslzchen) [reply osf/models/cedar_metadata.py:53]: oh, I mean grabbing the `metadata` in the post request from the FE in browser and verify it locally against the same template, but yeah, let's talk about it in the morning

**Review by adlius — APPROVED** [review]
I think it looks great and very straight forward. We need to verify this locally to make sure the CEDAR embeddable editor actually produces valid json that passes the JsonschemaValidator checks.

**Pull request closed without merging**